Merge Hyperdrive content into master

vapaunic opened the pull request:
This PR would merge the new hyper-parameter with Hyperdrive tuning code into the master.

pechyony commented on hyperparameter_tuning/CNN/README.md:
```diff
@@ -0,0 +1,3 @@
+# Multi-step forecasting with recurrent neural network that generates vector output
```
replace "recurrent" with "convolutional"

vapaunic replied:
Good catch, thanks Dmitry. I'll address these comments.

pechyony commented on hyperparameter_tuning/README.md:
```diff
@@ -0,0 +1,35 @@
+- Feed-forward network multi-step multivariate approach - [ff_multistep_config.json](ff_multistep_config.json)
+- RNN multi-step approach - [rnn_multistep_config.json](rnn_multistep_config.json)
+- RNN teacher forcing approach - [rnn_teacher_forcing_config.json](rnn_teacher_forcing_config.json)
+- RNN encoder decoder approach - [rnn_encoder_decoder_config.json](rnn_encoder_decoder_config.json)
```
Add pointer to JSON file of CNN

vapaunic commented:
@pechyony , I addressed your comments:
- Fixed up CNN readme wording, and the config file link
- added cell outputs in hyperdrive notebook.

Let me know if you have further comments, @angusrtaylor , @yijingchen , otherwise I'll merge the PR.
